src/liboslexec/runtimeoptimize.cpp

```diff
@@ -2353,10 +2353,9 @@ RuntimeOptimizer::resolve_isconnected ()
                 ASSERT (fieldsymid >= 0);
                 s = inst()->symbol(fieldsymid);
             }
-            if (s->connected())
-                turn_into_assign_one (op, "resolve isconnected() [1]");
-            else
-                turn_into_assign_zero (op, "resolve isconnected() [0]");
+            int val = (s->connected() ? 1 : 0) + (s->connected_down() ? 2 : 0);
```

Comment:
I guess it cannot be connected() and connected_down() at the same time. Maybe connected() should be connected_up(), but I'm not suggesting you rename the whole thing now. Maybe make this assignment a bit clearer like

val = s->connected() ? 1 : (s->connected_down() ? 2 : 0) : 0;

but I guess it's a matter of taste.